### PR TITLE
[ResourceCluster] Add basic host routes to v1 API.

### PR DESCRIPTION
**Pull request by Andyz26**
### Context

Add API for ResouceClustersHostActor public facing operations.

### Checklist

- [x] `./gradlew build` compiles code correctly
- [x] Added new tests where applicable
- [x] `./gradlew test` passes all tests
- [x] Extended README or added javadocs where applicable
- [x] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/api/akka/payloads/ResourceClustersPayloads.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.api.akka.payloads;
+
+public class ResourceClustersPayloads {
+    public static final String CLUSTER_ID = "mantisResourceClusterUT1";
+
+    public static final String RESOURCE_CLUSTER_CREATE = "{\"clusterId\":\"mantisResourceClusterUT1\",\"clusterSpec\":"
```
Comment (codyrioux) [inline mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/api/akka/payloads/ResourceClustersPayloads.java:22]: More of a personal preference than anything, but these may be painful to modify for testing if the data structure changes in any sort of meaningful way. Some kind of structure that gets serialized, or even a well formatted ".json" file that is read in would be easier to edit.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/ResourceClusterRouteHandler.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import io.mantisrx.control.plane.resource.cluster.proto.ListResourceClusterRequest;
+import io.mantisrx.control.plane.resource.cluster.proto.ProvisionResourceClusterRequest;
+import io.mantisrx.control.plane.resource.cluster.proto.ResourceClusterAPIProto.DeleteResourceClusterResponse;
+import io.mantisrx.control.plane.resource.cluster.proto.ResourceClusterAPIProto.GetResourceClusterResponse;
+import io.mantisrx.control.plane.resource.cluster.proto.ResourceClusterAPIProto.ListResourceClustersResponse;
+import io.mantisrx.control.plane.resource.cluster.proto.ScaleResourceRequest;
+import io.mantisrx.control.plane.resource.cluster.proto.ScaleResourceResponse;
+import java.util.concurrent.CompletionStage;
+
+public interface ResourceClusterRouteHandler {
```
Comment (codyrioux) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/ResourceClusterRouteHandler.java:29]: I may have missed some team discussion on this but do we have an official stance on interface names `I ResourceClusterRouteHandler` vs `ResourceClusterRouteHandler`? I don't have a strong opinion either way.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/control/plane/resource/cluster/resourceprovider/SimpleFileResourceClusterStorageProvider.java [hunk 3]
```diff
@@ -85,6 +87,28 @@ public CompletionStage<ResourceClusterSpecWritable> registerAndUpdateClusterSpec
...
+                        rc.getClusters().entrySet().stream()
+                        .filter(kv -> !Objects.equals(clusterId, kv.getKey()))
+                        .forEach(kv -> rcBuilder.cluster(kv.getKey(), kv.getValue()));
+                        return rcBuilder.build();
+                    })
+                )
+                .mapAsync(1, rc -> updateRegisteredClusters(rc))
+                .mapAsync(1, notUsed -> getRegisteredResourceClustersWritable())
+                .runWith(Sink.last(), system);
+        log.info("Return future on deregisterCluster: {}", clusterId);
```
Comment (codyrioux) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/control/plane/resource/cluster/resourceprovider/SimpleFileResourceClusterStorageProvider.java:108]: nit: Might be more appropriate to use `trace` or `debug` level here.

**Review by codyrioux — APPROVED** [review]
LGTM!

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/ResourceClusterRouteHandler.java (hunk 2 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/ResourceClusterRouteHandler.java:29]: This was brought up a few times in some of the previous PR comments from @sundargates IIRC and I am following his pattern in the new OSS mantis code at this point to be consistent.

**Comment by codyrioux** [comment]
I’ll make sure to follow that pattern in my code too to maintain
consistency. :)

File: mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/api/akka/payloads/ResourceClustersPayloads.java (hunk 1 above)
Comment (hmitnflx) [reply mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/api/akka/payloads/ResourceClustersPayloads.java:22]: +1

Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/api/akka/payloads/ResourceClustersPayloads.java]: +1. Moved to a resource JSON file.

**Pull request merged**